### ISSUE TITLE
Unit 2: YOLO chapter - Outline for the section

### ISSUE THREAD

**Issue by AntonisCSt** [sezan92/computer-vision-course#20]
Hey guys,

These are the proposed sections for the Yolo Section.

## Object Detection
- Connection with the previous section.
- How it is different from localization and classification?
- Sliding window approach (theoretically)
- R-CNN, Fast R-CNN (again theoretically not too many details)
- Outro → Methods still slow

## Yolo
- What it is and other introductory things about YOLO
-Mentioning Pros (Fast and very good mean average precision(mAP) compared to other state-of-the-art object detection) paper [[1](https://arxiv.org/pdf/2207.02696.pdf)]

### Architecture 
- Yolo difference with the previous method (SxS grid, bounding boxes) and CNN architecture
- Minmax expression and more details for the initial steps of Yolo
- Reference to Unit’s 6 how to evaluate object detection ([link](https://github.com/johko/computer-vision-course/blob/b70e4055a89f820de9de20bc777780face1685d1/chapters/en/Unit%206%20-%20Basic%20CV%20Tasks/object_detection.mdx#L31C1-L31C47))
- Yolo Architecture Intuition and what makes it so fast.
-Loss function

### Yolo over the years
How did it evolve?
Where is it today? Main differences

### Using Yolov8

The code part goes here to how it is used.
(Maybe we could add other cool stuff about Yolo here if it does not make it complicated)

### Tasks

- [x] #21 
- [x] #22 
- [x] #23 
- [x] #24 

## References

Open to suggestions for further improvement of the draft!

**Comment by sezan92** [comment]
> ## Object Detection
> * Connection with the previous section.
> * How it is different from localization and classification?
> * Sliding window approach (theoretically)
> * R-CNN, Fast R-CNN (again theoretically not too many details)
> * Outro → Methods still slow

I think  case of "Connection with the previous section", we can write "Connection with Convolutional Neural Network"

"How it is different from localization and classification -> how it is different from image classification"

"Previous methods, 
- sliding window
- RCNN , Fast RCNN, Faster RCNN, 
- "

**Comment by sezan92** [comment]
> ### Architecture
> * Yolo difference with the previous method (SxS grid, bounding boxes) and CNN architecture
> * Minmax expression and more details for the initial steps of Yolo
> * Reference to Unit’s 6 how to evaluate object detection ([link](https://github.com/johko/computer-vision-course/blob/b70e4055a89f820de9de20bc777780face1685d1/chapters/en/Unit%206%20-%20Basic%20CV%20Tasks/object_detection.mdx#L31C1-L31C47))
> * Yolo Architecture Intuition and what makes it so fast.
>   -Loss function

I think following order is better, 

"
1. Yolo , difference with the previous methods,
2. YOLO architecture intuition
3. Innovations -> (a) Loss function , (b) min-max expression 
4. Evaluation ,

**Comment by sezan92** [comment]
> ### Yolo over the years
> How did it evolve? Where is it today? Main differences
> 


Let us describe , differences/ advancements among Yolov1, Yolov2, YOLOV3, ....Yolov8

> ### Using Yolov8
> The code part goes here to how it is used. (Maybe we could add other cool stuff about Yolo here if it does not make it complicated)

For first PR I prefer writing how to use existing Yolov8 library , rather than building the architecture. it will make the PR faster, and we get feedbacks earlier.

**Comment by themurtazanazir** [comment]
How deep are we planning to go into different architectures? I think we should explain YOLOv1 pretty good and then write changes from previous versions for v2-v8

**Comment by sezan92** [comment]
> How deep are we planning to go into different architectures? I think we should explain YOLOv1 pretty good and then write changes from previous versions for v2-v8

I agree
